tachyon/crypto/commitments/mixed_matrix_commitment_scheme.h

```diff
@@ -0,0 +1,45 @@
+#ifndef TACHYON_CRYPTO_COMMITMENTS_MIXED_MATRIX_COMMITMENT_SCHEME_H_
+#define TACHYON_CRYPTO_COMMITMENTS_MIXED_MATRIX_COMMITMENT_SCHEME_H_
+
+#include <utility>
+#include <vector>
+
+#include "third_party/eigen3/Eigen/Core"
+
+#include "tachyon/crypto/commitments/mixed_matrix_commitment_scheme_traits_forward.h"
+#include "tachyon/math/matrix/matrix_types.h"
+
+namespace tachyon::crypto {
+
+template <typename Derived>
+class MixedMatrixCommitmentScheme {
+ public:
+  using Field = typename MixedMatrixCommitmentSchemeTraits<Derived>::Field;
+  using Commitment =
+      typename MixedMatrixCommitmentSchemeTraits<Derived>::Commitment;
+
+  [[nodiscard]] bool Commit(const std::vector<Field>& vector,
```

Comment:
Do you mean like this?

```suggestion
  [[nodiscard]] bool Commit(const std::vector<Field>& fields,
```

If so, I intentionally named it vector since we have this interface:

```c++
  [[nodiscard]] bool Commit(math::RowMajorMatrix<Field>&& matrix,
                            Commitment* result) {
```